src/lib/libast/string/base64.c

```diff
@@ -119,13 +118,12 @@ ssize_t base64encode(const void *fb, size_t fz, void **fn, void *tb, size_t tz,
         *tp++ = (fz == 2) ? m[(b >> 6) & 077] : PAD;
         *tp++ = PAD;
     }
-    if (n)
+    if (n) {
         n += (tp - tmp) - 1;
-    else {
-        if (tp > (unsigned char *)tb && *(tp - 1) == '\n') tp--;
+    } else {
+        if (tp != tmp && tp > (unsigned char *)tb && *(tp - 1) == '\n') tp--;
```

Comment:
Why are you comparing ` tp != tmp` ? I would appreciate a comment in source code here.

Reply:
I decided to turn it into an assert. Here's the comment I added:

```
// Static analysis warns that there is a theoretical path where the next `if` can be reached
// with tp == tmp. If that happens the `*(tp - 1)` would access the byte before the start of
// the tmp array.
```

It's due to this warning I put in the commit comment:

```
src/lib/libast/string/base64.c:124:51: The left operand of '==' is a garbage value
```

Basically this is yet another example of code that is poorly structured and plays fast and loose with the rules.